src/routers/StackRouter.js

```diff
@@ -101,23 +101,30 @@ export default (routeConfigs, stackConfig = {}) => {
       // Set up the initial state if needed
       if (!state) {
         let route = {};
-        if (
-          behavesLikePushAction(action) &&
-          childRouters[action.routeName] !== undefined
-        ) {
+        const childRouter = childRouters[action.routeName];
+        if (behavesLikePushAction(action) && childRouter !== undefined) {
+          let childState = {};
+          if (childRouter !== null) {
```

Comment:
we only execute this code if `childRouter !== undefined` (the above conditional), is it necessary to check that it's not null here? if so we should add a comment

Reply:
Sure I'll add a comment. This is the main fix from #3397. A child router is null if there is a route there but no router. Previously we had not been allowing the child router to handle the initial action